framework/source/class/qx/data/controller/ArrayFilter.js

```diff
@@ -0,0 +1,191 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2004-2009 1&1 Internet AG, Germany, http://www.1und1.de
+
+   License:
+     LGPL: http://www.gnu.org/licenses/lgpl.html
+     EPL: http://www.eclipse.org/org/documents/epl-v10.php
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * John Spackman (john.spackman@zenesis.com
+
+************************************************************************ */
+
+/**
+ * Manages a copy of a {@link qx.data.Array}, but with a filter and/or sort applied
+ * 
+ * The filtered/sorted array is available in the "filtered" property 
+ */
+qx.Class.define("qx.data.controller.ArrayFilter", {
+  extend: qx.core.Object,
+  
+  /**
+   * Constructor
+   */
+  construct: function() {
+    this.base(arguments);
+    this.setFiltered(new qx.data.Array);
+  },
+
+  /**
+   * Destructor
+   */
+  destruct: function() {
+    this._disposeArray(this.getFiltered());
+  },
+  
+  properties: {
+    /** Model - this is the source array that is filtered and sorted into the filtered property */
+    model: {
+      init: null,
+      nullable: true,
+      check: "qx.data.Array",
+      apply: "_applyModel",
+      event: "changeModel"
+    },
+    
+    /** Filtered and copied version of model; this can be changed, but not set to null */
+    filtered: {
+      nullable: false,
+      check: "qx.data.Array",
+      apply: "_applyFiltered",
+      event: "changeFiltered"
+    },
+    
+    /** Method called to sort */
+    sortMethod: {
+      init: null,
+      nullable: false,
+      apply: "_applyXxx"
+    },
+    
+    /** Method called to filter */
+    filterMethod: {
+      init: null,
+      nullable: true
```

Comment:
the filter method doesn`t get applied on change, I guess an apply: "_applyXxx" is missing here